### ISSUE TITLE
doesn't compile under texlive2019

### ISSUE THREAD

**Issue by casperdcl** [MIDL-Conference/MIDLLatexTemplate#3]
```
$ git log -n1
commit 5c0ca9bd52319393b23a3ef33d7fdf70bc15cb4e
Author: Tom Vercauteren <tom.vercauteren@kcl.ac.uk>
Date:   Fri Dec 13 08:41:08 2019 +0000

    recompiled
$ pdflatex --output-directory /tmp midl-sampleabstract.tex
pdflatex --output-directory /tmp midl-sampleabstract.tex 
This is pdfTeX, Version 3.14159265-2.6-1.40.20 (TeX Live 2019) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(./midl-sampleabstract.tex
LaTeX2e <2019-10-01> patch level 3
(./midl.cls
Document Class: midl 2018/10/09 v1.0 MIDL Proceedings
($HOME/.local/texlive/2019/texmf-dist/tex/latex/jmlr/jmlr.cls
Document Class: jmlr 2017/08/01 v1.24 (NLCT) Journal of Machine Learning Resear
ch
($HOME/.local/texlive/2019/texmf-dist/tex/latex/xkeyval/xkeyval.sty
($HOME/.local/texlive/2019/texmf-dist/tex/generic/xkeyval/xkeyval.tex
($HOME/.local/texlive/2019/texmf-dist/tex/generic/xkeyval/xkvutils.tex
($HOME/.local/texlive/2019/texmf-dist/tex/generic/xkeyval/keyval.tex))))
($HOME/.local/texlive/2019/texmf-dist/tex/latex/tools/calc.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/etoolbox/etoolbox.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/base/article.cls
Document Class: article 2019/10/25 v1.4k Standard LaTeX document class
($HOME/.local/texlive/2019/texmf-dist/tex/latex/base/size11.clo))
($HOME/.local/texlive/2019/texmf-dist/tex/latex/amsmath/amsmath.sty
For additional information on amsmath, use the `?' option.
($HOME/.local/texlive/2019/texmf-dist/tex/latex/amsmath/amstext.sty
($HOME/.local/texlive/2019/texmf-dist/tex/latex/amsmath/amsgen.sty))
($HOME/.local/texlive/2019/texmf-dist/tex/latex/amsmath/amsbsy.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/amsmath/amsopn.sty))
($HOME/.local/texlive/2019/texmf-dist/tex/latex/amsfonts/amssymb.sty
($HOME/.local/texlive/2019/texmf-dist/tex/latex/amsfonts/amsfonts.sty))
($HOME/.local/texlive/2019/texmf-dist/tex/latex/natbib/natbib.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/graphics/graphicx.sty
($HOME/.local/texlive/2019/texmf-dist/tex/latex/graphics/graphics.sty
($HOME/.local/texlive/2019/texmf-dist/tex/latex/graphics/trig.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/graphics-def/pdftex.def)))
 ($HOME/.local/texlive/2019/texmf-dist/tex/latex/url/url.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/xcolor/xcolor.sty
($HOME/.local/texlive/2019/texmf-dist/tex/latex/graphics-cfg/color.cfg)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/xcolor/x11nam.def))
($HOME/.local/texlive/2019/texmf-dist/tex/latex/algorithm2e/algorithm2e.st
y ($HOME/.local/texlive/2019/texmf-dist/tex/latex/base/ifthen.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/ifoddpage/ifoddpage.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/tools/xspace.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/relsize/relsize.sty))
($HOME/.local/texlive/2019/texmf-dist/tex/latex/jmlr/jmlrutils.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/hyperref/hyperref.sty
($HOME/.local/texlive/2019/texmf-dist/tex/generic/oberdiek/ltxcmds.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/generic/iftex/iftex.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/pdftexcmds/pdftexcmds.sty
($HOME/.local/texlive/2019/texmf-dist/tex/generic/infwarerr/infwarerr.sty)
) ($HOME/.local/texlive/2019/texmf-dist/tex/generic/oberdiek/kvsetkeys.sty
($HOME/.local/texlive/2019/texmf-dist/tex/generic/oberdiek/etexcmds.sty
($HOME/.local/texlive/2019/texmf-dist/tex/generic/iftex/ifluatex.sty)))
($HOME/.local/texlive/2019/texmf-dist/tex/generic/oberdiek/kvdefinekeys.st
y)
($HOME/.local/texlive/2019/texmf-dist/tex/generic/pdfescape/pdfescape.sty)
 ($HOME/.local/texlive/2019/texmf-dist/tex/latex/oberdiek/hycolor.sty
($HOME/.local/texlive/2019/texmf-dist/tex/latex/oberdiek/xcolor-patch.sty
($HOME/.local/texlive/2019/texmf-dist/tex/latex/oberdiek/hopatch.sty)))
($HOME/.local/texlive/2019/texmf-dist/tex/latex/letltxmacro/letltxmacro.st
y) ($HOME/.local/texlive/2019/texmf-dist/tex/latex/oberdiek/auxhook.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/kvoptions/kvoptions.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/hyperref/pd1enc.def)
($HOME/.local/texlive/2019/texmf-dist/tex/generic/oberdiek/intcalc.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/latexconfig/hyperref.cfg)
($HOME/.local/texlive/2019/texmf-dist/tex/generic/bitset/bitset.sty
($HOME/.local/texlive/2019/texmf-dist/tex/generic/oberdiek/bigintcalc.sty)
) ($HOME/.local/texlive/2019/texmf-dist/tex/generic/atbegshi/atbegshi.sty)
) ($HOME/.local/texlive/2019/texmf-dist/tex/latex/hyperref/hpdftex.def
($HOME/.local/texlive/2019/texmf-dist/tex/latex/atveryend/atveryend.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/rerunfilecheck/rerunfilech
eck.sty
($HOME/.local/texlive/2019/texmf-dist/tex/generic/oberdiek/uniquecounter.s
ty))) ($HOME/.local/texlive/2019/texmf-dist/tex/latex/hyperref/nameref.sty
($HOME/.local/texlive/2019/texmf-dist/tex/latex/oberdiek/refcount.sty)
($HOME/.local/texlive/2019/texmf-dist/tex/generic/oberdiek/gettitlestring.
sty)))) ($HOME/.local/texlive/2019/texmf-dist/tex/latex/mwe/mwe.sty)
(/tmp/midl-sampleabstract.aux)
($HOME/.local/texlive/2019/texmf-dist/tex/context/base/mkii/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/epstopdf-pkg/epstopdf-base
.sty
($HOME/.local/texlive/2019/texmf-dist/tex/latex/latexconfig/epstopdf-sys.c
fg)) (/tmp/midl-sampleabstract.out) (/tmp/midl-sampleabstract.out)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/amsfonts/umsa.fd)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/amsfonts/umsb.fd)
($HOME/.local/texlive/2019/texmf-dist/tex/latex/base/omscmr.fd)

Package natbib Warning: Citation `Lamport:Book:1989' on page 1 undefined on inp
ut line 81.


LaTeX Warning: Reference `fig:example' on page 1 undefined on input line 90.


LaTeX Warning: Reference `tab:example' on page 1 undefined on input line 91.


LaTeX Warning: Reference `eq:example' on page 1 undefined on input line 92.


Overfull \hbox (2.06706pt too wide) in paragraph at lines 88--93
[]\OT1/cmr/m/n/10.95 Note that the JMLR tem-plate pro-vides many handy func-tio
n-al-i-ties such as [][]\OT1/cmtt/m/n/10.95 \figureref
Runaway argument?
example-image.pdf \edef \Gin@ext {\Gin@temp }\fi \def \Gin@temp {.png\ETC.
! Paragraph ended before \@tempa was complete.
<to be read again> 
                   \par 
l.117 ...hics[width=0.5\linewidth]{example-image}}
                                                  
? 
```

**Comment by tvercaut** [comment]
I can't reproduce the error. Did you modify the template in any way?

**Comment by casperdcl** [comment]
No modifications. Fresh clone on Ubuntu 16.04 with a fresh full install of `texlive2019`.

**Comment by tvercaut** [comment]
I only have access to my mac so can't directly test your configuration.

I however just setup a basic CI integration for Ubuntu 16.04 Xenial and the build passed:
https://github.com/MIDL-Conference/MIDLLatexTemplate/runs/348729048
See the configuration here:
https://github.com/MIDL-Conference/MIDLLatexTemplate/blob/2c76a38ea1831c8bd3cae8d319dacd60ce3926b2/.travis.yml#L1-L6

Is `texlive-full` a different package than `texlive2019` on ubuntu?

For the record, I also cloned the repo in overleaf and it works fine there.

**Comment by casperdcl** [comment]
hmm I used https://tug.org/texlive/acquire-netinstall.html (i.e. http://mirror.ctan.org/systems/texlive/tlnet/install-tl-unx.tar.gz). I suppose I could use a docker container for compiling but the issue seems odd.

**Comment by tvercaut** [comment]
I don't have much ground to explore your particular issue. The CI now covers 3 OSes and 3 versions of texlive, all of which are building:
https://travis-ci.com/MIDL-Conference/MIDLLatexTemplate

I am closing this issue as it may be related to your instalation.

**Comment by casperdcl** [comment]
@tvercaut I can confirm the issue persists under `texlive-2019`. This is the default version currently installed using the recommended method (https://tug.org/texlive/acquire-netinstall.html) so it seems like an issue. It's likely upstream but I don't really have time to debug it.

In the meantime I can use `texlive-2017`. Let me know if anyone can compile under `texlive-2019` or `2020`.

**Comment by tvercaut** [comment]
I use texlive 2019 under mac without issue. This is reflected in travis CI. I was able to reproduce a similar issue to yours when using a miktex docker. It seems that an update (through `tlmgr` I guess in ubuntu) should fix your issues. What you get looks similar to https://github.com/latex3/latex2e/issues/191

**Comment by tvercaut** [comment]
Note that your issue doesn't seem specific to our templates. You can try this mwe which also produces the bug with a miktex docker:
```latex
\documentclass[pmlr]{jmlr}

\usepackage{mwe} % to get dummy images

\title[Short Title]{Full Title of Article}

\begin{document}

\begin{figure}[htbp]
 % Caption and label go in the first argument and the figure contents
 % go in the second argument
\floatconts
  {fig:example}
  {\caption{Example Image}}
  {\includegraphics[width=0.5\linewidth]{example-image.pdf}}
\end{figure}

The  JMLR template provides many handy functionalities such as \verb|\figureref| to refer to a figure,
e.g. \figureref{fig:example}.

\end{document}
```

**Comment by tvercaut** [comment]
@nlessmann Sorry I missed your PR as I was mostly looking at this in between meetings. Nice to see there is a workaround.

Feel free to resolve the conflicts I introduced and push directly to the master branch as I don't have much time to work on this.

**Comment by nlessmann** [comment]
Will do. Didn't actually see this issue, only saw that there was a typo in the travis file and tried to fix that. But this is indeed exactly the error that showed up.

**Comment by casperdcl** [comment]
#5 fixes this for me. Note that `tlmgr update --all` doesn't fix this issue as of now (ubuntu texlive 2019).

**Comment by tvercaut** [comment]
It looks like this has been fixed upstream in version 1.25 of the jmlr package:
https://ctan.org/pkg/jmlr?lang=en
https://ctan.org/ctan-ann/id/mailman.3352.1579371965.2586.ctan-ann@ctan.org
